### PR TITLE
Cleanup and fix various things

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Cleanup and fix various things

Clean up prototype code for direct google auth
Clean up loading auth2 api as it was interfering with setToken
Update setting token and user object from google auth
Implement signout reset

Stub for popup redirect

getUser from library and no longer silent authenticate
add flag for silent authentication

[stage-19]

## Motivation and Context
Fixes for the main PR.

## How Has This Been Tested?
Tested changes locally.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No.

### CODE REVIEW

**Comment by alex-deaconu** [comment]
Reverts a lot of the hacks from the previous commits. Cleans up a lot of code and implements some new features.

@ezequielc @santiagonoguez please review. Thanks!

File: web/scripts/components/userstate/app.js [hunk 1]
```diff
@@ -44,55 +42,23 @@
...
-          .when('/', ['$location', 'customAuthFactory', '$http', 'openidConnect',
-            function ($location, customAuthFactory, $http, openidConnect) {
+          .when('/', ['$location', 'userAuthFactory', 'openidConnect',
+            function ($location, userAuthFactory, openidConnect) {
               var hash = $location.hash();
 
-              if (hash && hash.match(/.*id_token=.*/)) {
+              if ($location.search().code || 
+                (hash && (hash.match(/.*id_token=.*/) || hash.match(/access_token=.*/)))) {
                 var idToken = hash.split('&')[1].split('=')[1];
```
Comment (santiagonoguez) [inline web/scripts/components/userstate/app.js:51]: @alex-deaconu I don't completely get this simplification. if $location.search().code returns truthy here, it is ensured that has is also truthy so the call below to hash.split won't fail with a null pointer error ?

Comment (alex-deaconu) [reply web/scripts/components/userstate/app.js:51]: @santiagonoguez good catch, I only left this for logging purposes, I will correct in my next PR. Thanks.

Comment (alex-deaconu) [reply web/scripts/components/userstate/app.js]: @santiagonoguez actually, I've fixed this via the latest commit. Thanks.

**Review by santiagonoguez — APPROVED** [review]
@alex-deaconu I can't say I completely follow all of the changes, but I tested again scenarios in stage-19 and everything looks fine, save for the details we already discussed, thanks !

**Comment by alex-deaconu** [comment]
@santiagonoguez sounds good, still a work in progress. Let me know if there's any concerns you would like me to address. Thanks.

**Pull request merged**